test/externalTests/gnosis-v2.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2020 solidity contributors.
+#------------------------------------------------------------------------------
+source scripts/common.sh
+source test/externalTests/common.sh
+
+verify_input "$1"
+SOLJSON="$1"
+
+function install_fn { npm install --package-lock; }
+function compile_fn { npx truffle compile; }
+function test_fn { npm test; }
+
+function gnosis_safe_test
+{
+    OPTIMIZER_LEVEL=1
+    CONFIG="truffle-config.js"
+
+    truffle_setup "$SOLJSON" https://github.com/solidity-external-tests/safe-contracts.git v2_070
+
+    sed -i 's|github:gnosis/mock-contract#sol_0_5_0|github:solidity-external-tests/mock-contract#master_070_new|g' package.json
+    sed -i -E 's|"@gnosis.pm/util-contracts": "[^"]+"|"@gnosis.pm/util-contracts": "github:solidity-external-tests/util-contracts#solc-7"|g' package.json
+    rm -f package-lock.json
+    rm -rf node_modules/
```

Comment:
These two lines were added as a hacky fix to some problems with node while running tests. Not sure if they are still relevant.

Reply:
Good point. They shouldn't be. These files do not even exist before `run_install` anyway.

I remember that I was confused why it's even being one and I was going to remove it but I could not find it later because I assumed that I saw that part in `common.sh` :) I'll remove it now.

Reply:
Actually, `package-lock.json` was in Gnosis repo before I pulled in upstream changes and it's generally a good practice to commit it as it pins versions of all the transitive dependencies. Here it probably interfered with the change that forced a different Truffle version. That's no longer needed so we can safely remove it.

Reply:
Here is the PR that led to this change: https://github.com/ethereum/solidity/pull/9493

Reply:
Thanks. Yeah, so that's what I assumed. Removing `package.json` was the right thing to do in that situation (it by design overrides `package.json` and is meant o be used only if you want to reproduce the same exact dependency tree the upstream had).

Reply:
Done. I got rid or removing `node_modules` but I actually left the removal of `package-lock.json` in, at least in those scripts that do modify `package.json`. This makes them a bit more robust because even if the file is not there now, it could always get added to the repo later. I added a comment explaining why it's being doe.